equinix/resource_metal_device_acc_test.go

```diff
@@ -111,36 +101,42 @@ data "equinix_metal_plans" "test" {
-    //With these operations we use current time seconds as the seed, and avoid using a third party provider in the Equinix provider tests
-    plans             = data.equinix_metal_plans.test.plans
-    plans_random_num  = formatdate("s", timestamp())
-    plans_length      = length(local.plans)
-    plans_range_limit = ceil(59 / local.plans_length) == 1 ? local.plans_length : 59
-    plan_idxs         = [for idx, value in range(0, local.plans_range_limit, ceil(59 / local.plans_length)) : idx if local.plans_random_num <= value]
-    plan_idx          = length(local.plan_idxs) > 0 ? local.plan_idxs[0] : 0
-    plan              = local.plans[local.plan_idx].slug
-
-	//Operations to select a facility randomly and avoid race conditions with facilities without capacity.
-    //With these operations we use current time seconds as the seed, and avoid using a third party provider in the Equinix provider tests
-    facilities             = tolist(setsubtract(local.plans[local.plan_idx].available_in, ["nrt1", "dfw2", "ewr1", "ams1", "sjc1", "ld7", "sy4", "ny6"]))
-    facilities_random_num  = formatdate("s", timestamp())
-    facilities_length      = length(local.facilities)
-    facilities_range_limit = ceil(59 / local.facilities_length) == 1 ? local.facilities_length : 59
-    facility_idxs          = [for idx, value in range(0, local.facilities_range_limit, ceil(59 / local.facilities_length)) : idx if local.facilities_random_num <= value]
-    facility_idx           = length(local.facility_idxs) > 0 ? local.facility_idxs[0] : 0
-    facility               = local.facilities[local.facility_idx]
-
-    //Operations to select a metro randomly and avoid race conditions with metros without capacity.
-    //With these operations we use current time seconds as the seed, and avoid using a third party provider in the Equinix provider tests
-    metros             = tolist(local.plans[local.plan_idx].available_in_metros)
-    metros_random_num  = formatdate("s", timestamp())
-    metros_length      = length(local.metros)
-    metros_range_limit = ceil(59 / local.metros_length) == 1 ? local.metros_length : 59
-    metro_idxs         = [for idx, value in range(0, local.metros_range_limit, ceil(59 / local.metros_length)) : idx if local.metros_random_num <= value]
-    metro_idx          = length(local.metro_idxs) > 0 ? local.metro_idxs[0] : 0
-    metro              = local.metros[local.metro_idx]
+    // Select a random plan
+	selected_plan     = data.equinix_metal_plans.test.plans[random_integer.plan_idx.result]
+    plan              = local.selected_plan.slug
+
+	// Select a random facility from the facilities in which the selected plan is available, excluding decommed facilities
+    facilities             = tolist(setsubtract(local.selected_plan.available_in, ["nrt1", "dfw2", "ewr1", "ams1", "sjc1", "ld7", "sy4", "ny6"]))
+    facility               = local.facilities[random_integer.facility_idx.result]
```

Comment:
curious if this is still in use

Reply:
@displague I found some `local.facilities` refs in tests, so still in use https://github.com/equinix/terraform-provider-equinix/pull/329/files#diff-175a9bc69732939f0bf4dcedd034f66e05957561ffee186258e42fba19c2d2bcL62